### DISCUSSION TITLE
Where should we keep the code?

### DISCUSSION THREAD

**Discussion by JohnG-cc** [CredComSoc/ProtocolWorkingGroup#7] [General]
I imagine this working group is going to generate Issue tickets and pull requests... 

Where should that be happening? Should there be another copy of the code here on GitHub under CredComSoc? Should this be co-maintained with the GitLab copy given that we can push and pull between them? Or are we wanting to settle on a single repo, either the current GitLab one or a new GitHub one, and manage issues and pull requests through there?

@matslats @dilgreen @utunga @tfwoodroofMCS  what do you think?

**Comment by matslats** [comment]
I'm pushing my code to gitlab right now.
Happy to hear other proposals.

**Comment by utunga** [comment]
Context: Had a zoom call with @dilgreen yesterday on which this topic was discussed. Also it was touched on in the discussion John Gray and I had on the weekend. 

The upshot of that is that I'm pretty sure we're gonna be moving the Credit Co-Op Tech repo here (ie to github/CredComSoc). This will also be an opportunity to 'open it up' in the sense of making it public, inviting public requests, adding a CONTRIBUTING.md etc. 

However that doesn't by any means we need to move GitLab/Credit-Commons code... https://gitlab.com/credit-commons If it was up to me I'd probably even err on the side of _leaving Credit-Commons (and the cc-node code in particular) where it is._  

- I think there is a case to be made that **gitlab** is less (likely to be) evil that github aka Microsoft. 
- I there is also a pretty solid case to be made that the collab tools, ease of use, planning etc are more approachable at **github** than gitlab.
- But most of all I think there is a case to be made for **diversity** 

By having repos in more than one place we remind ourselves of the fact that every git copy of the repo is a first class copy - whether it is on our machine or one of the cloud hosts. Also if we get to the point where collaborators need to be in one or the other org to collaborate there may be some overlap and some not. This means if we do have to, lets say, move away from github it won't be such a drama to do.

But yeah not too fussed if others have strong opinions about bringing it all together into one place (either now or later on) or alternatively feel that we should use gitlab over github.

**Reply by JohnG-cc** [reply]
Works for me...

**Reply by dilgreen** [reply]
I have to do some work to get gitlab sorted - at present we have been subjected to a'read-only' status as we have not achieved 'open source org' status.

Agree with @utunga

**Comment by bu6n** [comment]
I think I already mentioned that, but it would be good if both gitlab and github organizations had the name `credit-commons`.
Changing the github organization from `CredComSoc` to `credit-commons` should not be a problem, since GitHub redirects the repos' links until someone registers an organization with the same name and a repo with the same name. I would suggest to:

1. Come to a consensus about the name change (I suggest that @dilgreen, @tfwoodroofMCS, @utunga and @JohnG-cc put a thumbs up emoji on this comment if they agree with the name change) 
2. Sync with someone who has the rights to rename `CredComSoc`. I currently have the rights on the `credit-commons` organization, so I should first delete it so that the owner of the `CredComSoc` org can rename it to `credit-commons`. 
I'm happy to do the change myself if someone wants to give me temporary admin rights in the `CredComSoc` organization, but I would understand if that's not the case.
3. Register a new organization called `CredComSoc` so that no one else overwrites the paths of the old repo
4. Recommend that developers change their git remotes (and other links if existing)

Any thoughts?
